src/naive/datetime/mod.rs

```diff
@@ -126,6 +126,27 @@ impl NaiveDateTime {
         let datetime = NaiveDateTime::from_timestamp_opt(secs, nsecs);
         datetime.expect("invalid or out-of-range datetime")
     }
+    /// Creates a new [NaiveDateTime] from milliseconds since the UNIX Epoch.
+    ///
+    /// The UNIX epoch starts on midnight, January 1, 1970, UTC.
+    ///
+    /// Returns `None` on an out-of-range number of milliseconds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chrono::NaiveDateTime;
+    /// let timestamp_millis: i64 = 1662921288; //Sunday, September 11, 2022 6:34:48 PM
+    /// let naive_datetime = NaiveDateTime::from_timestamp_millis(timestamp_millis);
+    /// assert!(naive_datetime.is_some());
+    /// assert_eq!(timestamp_millis, naive_datetime.unwrap().timestamp_millis());
+    /// ```
+    #[inline]
+    pub fn from_timestamp_millis(millis: i64) -> Option<NaiveDateTime> {
+        let secs = millis / 1000;
+        let nsecs = (millis % 1000) as u32 * 1_000_000;
```

Comment:
this looks problematic as the input `millis` could be negative, and will still be negative after taking the modulo ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7e7ecdb063e7200a2970e92632ff1fbc))

One option would be to have two cases, depending on whether `millis` is positive or negative, similar to the logic used [here](https://github.com/chronotope/chrono/blob/e4911675b9c9cf6f9dc33015987494843b35fee8/src/naive/time/mod.rs#L535-L596)

The other option is just:
`NaiveDate::from_ymd(1970,1, 1).and_hms(0, 0, 0) + TimeDelta::from_millis(millis)`


Also it would be preferable to use `TryFrom` instead of `as`

Reply:
Awesome, let's deal with that. I'm super busy ATM, please give me by the end of the week :)

Reply:
I don't think we should go through `TimeDelta` here, which adds a bunch of overhead and feels like a layering violation. Would be better to do a minimal calculation and still defer to `NaiveDateTime::from_timestamp_opt()`.

Reply:
Good call RE. avoiding `TimeDelta` especially as `from_millis` may well get deprecated anyway

Reply:
Done. The nanos calculation is split, as when the timestamp in millis is negative, an additional `abs()` op is done.